include/class.forms.php

```diff
@@ -2274,6 +2274,30 @@ function render($options=array()) {
     }
 }
 
+
+class TextboxSelectionWidget extends TextboxWidget {
+    //TODO: Support multi-input e.g comma separated inputs
+    function render($options=array()) {
+
+        if ($this->value && is_array($this->value))
+            $this->value = current($this->value);
+
+        parent::render($options);
+    }
+
+    function getValue() {
+
+        $value = parent::getValue();
+        if (($i=$this->field->getList()->getItem((string) $value)))
```

Comment:
Just a thought here. Should we also consider content in the `extra` column so abbreviations would be welcome in the textbox too?